jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/util/JaxrsUtil.java

```diff
@@ -34,6 +35,16 @@
 
 public abstract class JaxrsUtil {
 
+  public static final int unprivilegedPort() throws IOException {
+    int port = 0;
+    while (port < 1024) {
+      try (final ServerSocket serverSocket = new ServerSocket(0)) {
+        port = serverSocket.getLocalPort();
+      }
+    }
```

Comment:
This loop has the potential of being infinite, in theory. We should cap the number retries.

Reply:
I doubt that you will be able to actually detect that endless loop in reality (can you proof your claim?), but I can add a cap at 1025 tries (1 more than the size of the privileged range).

Reply:
You can't easily prove it either way, that's the point.